src/MiniCover/Reports/XmlReport.cs

```diff
@@ -55,18 +58,31 @@ public static void Execute(InstrumentationResult result, string output, float th
 
                         var methodPoints = instruction.Select(methodPoint =>
                         {
-                            var hitCount = hitInstructions.Count(hit => hit.Equals(methodPoint));
+                            
+                            var counter = hits.VisitedForMethod(methodPoint.Id);
 
-                            return new XElement(
+                            var point = new XElement(
                                 XName.Get("seqpnt"),
-                                new XAttribute(XName.Get("visitcount"), hitCount),
+                                new XAttribute(XName.Get("visitcount"), counter),
                                 new XAttribute(XName.Get("line"), methodPoint.StartLine),
                                 new XAttribute(XName.Get("column"), methodPoint.StartColumn),
                                 new XAttribute(XName.Get("endline"), methodPoint.EndLine),
                                 new XAttribute(XName.Get("endcolumn"), methodPoint.EndColumn),
                                 new XAttribute(XName.Get("excluded"), "false"),
                                 new XAttribute(XName.Get("document"), Path.Combine(result.SourcePath, file.Key))
-                            );
+                                );
+                            var methodHits = hits.ForMethod(methodPoint.Id).ToArray();
+
+                            var tests = methodHits.OfType<Hit.HitWithTestsInformation>().Select(h => new XElement(XName.Get("tests"),
```

Comment:
I know that report is named just XMLReport, but it was created to generate XML in NCover format:
https://github.com/lucaslorentz/minicover/pull/7

Is the tests element part of NCover XML format?

Maybe we should rename it to NCoverReport, or NCoverXMLReport and create a new XML report for MiniCover.

Reply:
oh didn't knew. ok doing that, btw I think we should maybe go to a json format with the saving at the end and have report object from which we can generate other format.

Reply:
No problem. There was no way to know 😄 I know because I merged the other PR.

I also prefer JSON if we create minicover own output format.

But maybe there is no need to create minicover own output now, not even an XML. Do you know any tool that would read the test methods from it? Maybe it's better to add test methods to HTML report.

What do you think?

Reply:
the json I mean to use it for the hitfile.
But I would like to have a Json format that I could read.

Reply:
On that case. I also like a more structured format for hits file.

But I think one requirement is that it should be a format that allows appending more hits without parsing/rewriting the whole file.

Because the user could run test commands multiple times before generating coverage reports.

Maybe multiple jsons, like:
```
{
  instructionId: '1'
  testMethod: 'test method'
}
{
  instructionId: '2'
  testMethod: 'test method'
}
```

Or maybe some binary format, cause with JSON it will be too big. Protobuf, flatbuffers, bson?

Reply:
I agree binary format would be far better, but I think we can go step by step on this one